### PR TITLE
Verifying ISO is downloaded completely

### PR DESCRIPTION
**Pull request by nshidlin**
In some cases the ISO does not download completely

This PR adds a retry to download the image if the size of the resultant image does not match the content length and raises a clear exception in the case the retry failed as well

### CODE REVIEW

**Comment by YuviGold** [comment]
What could cause it? Just a flaky network and on a 2nd try it would pass?

**Comment by nshidlin** [comment]
@YuviGold 
> What could cause it? Just a flaky network and on a 2nd try it would pass?

Yes, could be caused by a flaky network. I've seen a second try succeed while testing this change. If the second try fails, then at at least an exception is raised, failing the test on the iso download.

File: discovery-infra/test_infra/assisted_service_api.py [hunk 1]
```diff
@@ -114,9 +114,14 @@ def cluster_get(self, cluster_id):
...
 
-    def _download(self, response, file_path):
+    def _download(self, response, file_path, verify_file_size=False):
         with open(file_path, "wb") as f:
             shutil.copyfileobj(response, f)
+        if verify_file_size:
+            content_length = int(response.headers['content-length'])
+            actuall_file_size = os.path.getsize(file_path)
+            if actuall_file_size < content_length:
+                raise RuntimeError(f'Could not complete ISO download {file_path}')
```
[the remark this answers is not in the document]
Comment (nshidlin) [reply discovery-infra/test_infra/assisted_service_api.py]: done

**Review by YuviGold — APPROVED** [review]
LGTM

**Pull request merged**